### ISSUE TITLE
[5.7-beta][isolatedDeclarations][regression] Missing explicit type annotation false positive

### ISSUE THREAD

**Issue by MichaelMitchell-at** [microsoft/TypeScript#60192]
### 🔎 Search Terms

explicit type annotation false positive isolated declarations

### 🕗 Version & Regression Information

- This changed between versions 5.6.0 and 5.7.0-beta

### ⏯ Playground Link

https://www.typescriptlang.org/dev/bug-workbench/?target=7&noCheck=true&isolatedDeclarations=true&emitDeclarationOnly=true&ts=5.7.0-dev.20241010#code/PTAEAEGcBcCcEsDG0Bco4FcCmAoEEATLRAGwENYzp4B7AOzU133DpoGEALYga0dmx4w4LAFt40ACLFylavQDydEgE9+glvEg1y0LAWmkKVWnUjrmwsiRI0A7gClzoAGbXIliJE72AouOghLx87fwk9AgAxeBIsNDIAOgIE6EgcIPAXGKw6MlE40ETUnHhRAAcaWGhQAG8AFRUyrABfV1gaUVAAchcaGi6AbnSsAA8KqtBEehhXPtAAXlAACmhGrAB+NAamgEo0ADcaeAIFgD5a5qGgA

### 💻 Code

```ts
// @strict: true
// @declaration: true
// @noCheck: true
// @emitDeclarationOnly: true
// @isolatedDeclarations: true
// @allowJs: false
// @showEmit
// @showEmittedFile: a.d.ts

import {Type} from 'foo';

export const foo = (type?: Type): void => {};
                    ^^^^^^^^^^^ Parameter must have an explicit type annotation with --isolatedDeclarations.(9011)
```

### 🙁 Actual behavior

Error even though type annotation exists

### 🙂 Expected behavior

No error

### Additional information about the issue

Sorry, I don't have time at the moment to bisect to the commit that introduced the regression

**Comment by Andarist** [comment]
Based on some other recent issues I was looking at - this should report an error about lack of `| undefined` in the annotation.

**Comment by MichaelMitchell-at** [comment]
> Based on some other recent issues I was looking at - this should report an error about lack of `| undefined` in the annotation.

Interesting. Why is that so? Is that only when `exactOptionalPropertyTypes` is off?

**Comment by RyanCavanaugh** [comment]
Seems like a bug?

EOPT doesn't have any effect on parameters FWIW

**Comment by Andarist** [comment]
Ah, it might be that the `| undefined` is required only when dealing with parameter properties (or however those are called). See https://github.com/microsoft/TypeScript/issues/60123 and https://github.com/microsoft/TypeScript/pull/60129

FWIW, this case here is likely also related to an unresolved type that is actually an error type (a special kind of `any`)
